idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+				subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+				iarr[subscripts]=0
+			endif else begin
+
+;====create a simple centre point======
+				arccent0=coord[0,(cent[0]),0]
+				arccent1=coord[1,0,(cent[1])]
+				
+;=====finds coordinates of CH boundaries=======
+				Xwb=coord[0,(max(xy[0,info(i).offset+chpts])),0]
+				Ywb=coord[1,(max(xy[0,info(i).offset+chpts])),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]]
+				Xeb=coord[0,(min(xy[0,info(i).offset+chpts])),0]
+				Yeb=coord[1,(min(xy[0,info(i).offset+chpts])),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq min(xy[0,info(i).offset+chpts])))]]
+				Ynb=coord[1,0,(max(xy[1,info(i).offset+chpts]))]
+				Xnb=coord[0,xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq max(xy[1,info(i).offset+chpts])))],(max(xy[1,info(i).offset+chpts]))]
+				Ysb=coord[1,0,(min(xy[1,info(i).offset+chpts]))]
+				Xsb=coord[0,xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq min(xy[1,info(i).offset+chpts])))],(min(xy[1,info(i).offset+chpts]))]
+
+				eastl=lon[where(coord[0,*,0] eq xeb),where(coord[1,0,*] eq yeb)]
+				westl=lon[where(coord[0,*,0] eq xwb),where(coord[1,0,*] eq ywb)]
+				width=westl-eastl
+
+;====classifies off limb CH regions========
+				if (arccent0^(2)+arccent1^(2)) gt rs^(2) or (coord[0,xy(0,INFO(I).OFFSET),0]^(2) + (coord[1,0,xy(1,INFO(I).OFFSET)])^(2)) gt rs^(2) then begin
+					subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+					offarr[subscripts]=1
+				endif else begin
+
+;=====classifies on disk coronal holes=======
+					subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+					iarr[subscripts]=ident
+
+;====create an array for magnetic polarity
+					poslin=where(iarr eq ident)
+					pos=fltarr(n_elements(poslin),2)
+					conver=(s[1]/2)*dattoarc/hin.cdelt1-(s[2]/2)
+					convermul=dattoarc/hin.cdelt1
+					pos[*,0]=((poslin/float(s[1]))-(poslin/s[1]))*s[1]  
+					pos[*,1]=poslin/s[1]       
+					pos[*,0]=(pos[*,0]-(s[1]/2))*convermul+(s[2]/2) 
+					pos[*,1]=(pos[*,1]-(s[1]/2))*convermul+(s[2]/2) 
+					npix=histogram(hd[pos[*,0],pos[*,1]],binsize=1)
+					magpol=indgen(max(hd[pos[*,0],pos[*,1]])-min(hd[pos[*,0],pos[*,1]]))+min(hd[pos[*,0],pos[*,1]])
+
+					wh=where(npix eq 0)
+					if wh[0] ne -1 then npix[where(npix eq 0)]=1
+
+;=====magnetic cut offs dependant on area=========
+					if abs((total(npix[where(magpol gt 0)])-total(npix[where(magpol lt 0)]))/sqrt(total(npix))) ge 10 or arcar gt 9000 then begin 
+						if abs(mean(hd[pos[*,0],pos[*,1]])) gt garr[cent[0],cent[1]] or arcar gt 40000 then begin
+
```

Comment:
Two `if` blocks to invert, do the opossite and `continue` the loop.
if `iarr` starts with all the values as `0` then you can clean all this indentation.